components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -14,7 +14,27 @@
 #include "brave/components/brave_wallet/browser/eth_json_rpc_controller.h"
 #include "brave/components/brave_wallet/browser/eth_response_parser.h"
 #include "brave/components/brave_wallet/common/web3_provider_constants.h"
+#include "components/grit/brave_components_strings.h"
 #include "components/user_prefs/user_prefs.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace {
+
+void RespondForEthereumChainRequest(
+    brave_wallet::BraveWalletProviderImpl::AddEthereumChainCallback callback,
+    const std::string& error) {
+  base::flat_map<std::string, std::string> headers;
+  if (!error.empty()) {
+    auto response = base::StringPrintf(
+        brave_wallet::kJsonResponseAddEthereumChainError, error.c_str());
+    std::move(callback).Run(200, response, headers);
```

Comment:
how are we going to propagate the error? If you pass 200 here, `BraveWalletJSHandler::OnRequest` would only call `ALLOW_UNUSED_LOCAL(resolver->Resolve(context, result));`.
See `OnEthereumPermissionRequested` and `OnGetAllowedAccounts`, it calls `ALLOW_UNUSED_LOCAL(resolver->Reject(context, result))` when `!success`

Reply:
I think we should reject the promise with error in failed cases.
See https://docs.metamask.io/guide/rpc-api.html#other-rpc-methods for metamask usage example.

Reply:
In this case, I think it's probably better to just have `OnAddEthereumChain` in `BraveWalletJSHandler` instead of sharing `OnRequest` and make weird accommodations. For example, http_code doesn't make any sense for AddEthereumChain API as this doesn't need to be sent to remote server.